docs/guide/headless-cms.md

```diff
@@ -0,0 +1,177 @@
+
+# A single page with document _id `903y5n76e8j:en:published`
+GET https://example.rocks/api/v1/@apostrophecms/page/903y5n76e8j:en:published?apikey=000apikey123
+```
+
+The module name used for these requests is always `@apostrophecms/page`, the module that manages pages in Apostrophe. The response for a "get one" request is also similar to a single piece `GET` request. It is a single page object with the page data properties immediate at the top level.
+
+The response for page "get all" requests work differently by default. This is because it reflects that pages have a hierarchical structure. The home page is the "parent" of the top-level pages, which then have additional "child" pages, and so on. As such, the response to a "get all" route as written above looks something like this:
+
+```json
+{
+    "_id": "ckhdscx5900054z9k88uqs16w",
+    "title": "Home Page",
+    "slug": "/",
+    // ... additional properties
+    "_url": "http://localhost:3000/",
+    "_ancestors": [],
+    // Child pages 👇
+    "_children": [
+        {
+            "_id": "ckhfen8ls0005mq9k8p9izkjt",
+            "title": "Blog",
+            "slug": "/blog",
+            // ... additional properties
+            "_url": "http://localhost:3000/blog",
+            "_children": []
+        },
+        {
+            "_id": "ckhurshqd00088v9k6pfnqpjz",
+            "title": "About us",
+            "slug": "/about-us",
+            // ... additional properties
+            "_url": "http://localhost:3000/about-us",
+            "_children": []
+        }
+    ]
+}
+```
+
+The home page object is returned with a `_children` array with the first-level pages.
+
+**We can request this data as a flat array** by adding the `?flat=1` query parameter.
+
+```
+GET https://example.rocks/api/v1/@apostrophecms/page?flat=1&apikey=000apikey123
+```
+
+**And expand the results to include all pages, regardless of page tree depth** by using the `?all=1` query parameter. This also works with the nested object format.
+
+```
+GET https://example.rocks/api/v1/@apostrophecms/page?flat=1&all=1apikey=000apikey123
+```
+
+## Getting rendered HTML in responses
+
+Using an object of JSON-like data from a REST API is very normal from a headless CMS. Traditionally the application's "head" (such as a single page Vue.js, React, or Angular application) would use that structured data to display product, event, or similar content information. It is less common for a headless CMS to deliver page content that is tied to particular HTML markup and layout. More often, landing page text and images might be coded directly into the app code or otherwise not be from the same data source.
+
+Apostrophe support both approaches. The in-context editing features and [widget areas](/guide/areas-and-widgets.md) give editors more control over content, but in it's default form areas are fairly incomprehensible data objects. It is when they are rendered with the widget template files that they are fully realized.
```

Comment:
Apostrophe supports both approaches. The in-context editing features and [widget areas](/guide/areas-and-widgets.md) give editors more control over content, but in its default form areas are fairly incomprehensible data objects. It is when they are rendered with the widget template files that they are fully realized.

Reply:
I corrected some basic grammar/typo stuff above. I think the description of this could use some reworking in future iterations, but OK with this for now.